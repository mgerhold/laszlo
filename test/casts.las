function split(string: String, delimiter: Char) ~> [String] {
    let result = [];
    let current = "";
    for char in string {
        if char == delimiter {
            result += [current];
            current = "";
            continue;
        }
        current = current + char;
    }
    if current.length > 0 {
        result += [current];
    }
    return result;
}

let s = "100
200";
let parts = split(s, '\n');
for part in parts {
    let number = part => I32;
    let result = number * 2;
    println(number + " * 2 = " + 2 * number);
    let string_result = result => String;
    println(string_result + " (" + typeof(string_result) + ")");
}

assert(true => String == "true");
assert(false => String == "false");
assert(42 => String == "42");
assert('!' => String == "!");
assert((1..10) => String == "1..10");
