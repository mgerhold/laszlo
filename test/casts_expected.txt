100 * 2 = 200
200 (String)
200 * 2 = 400
400 (String)